.github/actions/setup-python/action.yml

```diff
@@ -0,0 +1,20 @@
+name: Setup Python and uv
+inputs:
+  skip-python-setup:
+    required: false
+    description: "Whether to skip installing python using Github's `setup-python` action"
+    default: false
+runs:
+  using: "composite"
+  steps:
+    # Use the setup-python action to take advantage of the cache. uv will
+    # symlink to this version.
+    - name: Setup system python
+      if: ${{ inputs.skip-python-setup != 'true' }}
+      uses: actions/setup-python@v5
+      with:
+        python-version-file: '.python-version'
```

Comment:
Are you sure we need to manually run the Python setup here? The documentation for the GitHub action doesn't mention this and it seems there is a cache option: https://github.com/astral-sh/setup-uv?tab=readme-ov-file#enable-caching.

Reply:
The [`uv` documention ](https://docs.astral.sh/uv/guides/integration/github/#setting-up-python) states that it is faster to use the official `setup-python` Github Action:

> Alternatively, the official GitHub setup-python action can be used. This can be faster, because GitHub caches the Python versions alongside the runner.

The cache option uses the Github Action's cache which is slower and also needs a `uv.lock` file to be checked into the repository with all the dependencies resolved (like Cargo.lock) for handling cache invalidation. We don't do that yet, but could be done in a follow up PR. This PR is to have the basic support in place to fix #34095.

Reply:
Also I should note that the cache option in `setup-uv` action refers to the [caching of python dependencies](https://docs.astral.sh/uv/concepts/cache/) and not caching of the interpreter binaries. While we could benefit from caching dependencies, as mentioned in the previous comment, it requires a 'uv.lock' or some other strategy to invalidate the cache when the dependencies change. This can be explored in a followup PR.